### PR TITLE
libunwind: fix for aarch64 and non-4K pages

### PR DESCRIPTION
**Pull request by tpwrules**
###### Description of changes

This adds an upstream patch which fixes libunwind for aarch64 systems with 16K or 64K pages. Although such systems are quite rare, the X server crashes on startup without this patch, so adding it is vital to a graphical NixOS experience.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libunwind/default.nix [hunk 1]
```diff
@@ -9,6 +9,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-SmrsZmmR+0XQiJxErt6K1usQgHHDVU/N/2cfnJR5SXY=";
   };
 
+  # Fix for aarch64 and non-4K pages. Remove once upgraded past 1.6.2.
+  patches = lib.optional stdenv.isAarch64 (fetchpatch {
+    url = "https://github.com/libunwind/libunwind/pull/330.patch";
```
Comment (Artturin) [inline pkgs/development/libraries/libunwind/default.nix:14]: this is not a reproducible url. someone could push to the pull request and change it

go to the latest commit in the pull and add `.patch` to the end

Comment (tpwrules) [reply pkgs/development/libraries/libunwind/default.nix]: Good point. thanks for noticing

File: pkgs/development/libraries/libunwind/default.nix [hunk 2]
```diff
@@ -9,6 +9,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-SmrsZmmR+0XQiJxErt6K1usQgHHDVU/N/2cfnJR5SXY=";
   };
 
+  # Fix for aarch64 and non-4K pages. Remove once upgraded past 1.6.2.
+  patches = lib.optional stdenv.isAarch64 (fetchpatch {
+    url = "https://github.com/libunwind/libunwind/commit/cf6b9aadd60a85107afe5196251ca2ed198a96d0.patch";
+    sha256 = "sha256-z3Hpg98D4UMmrE/LC596RFcyxRTvDjD4k7llDPfz1NI=";
+  });
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libunwind/default.nix:16]:
```suggestion
  patches = [
    (fetchpatch {
      url = "https://github.com/libunwind/libunwind/commit/cf6b9aadd60a85107afe5196251ca2ed198a96d0.patch";
      sha256 = "sha256-z3Hpg98D4UMmrE/LC596RFcyxRTvDjD4k7llDPfz1NI=";
    })
  ];
```
Prevent bit rotting.

File: pkgs/development/libraries/libunwind/default.nix [hunk 3]
```diff
@@ -9,6 +9,14 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-SmrsZmmR+0XQiJxErt6K1usQgHHDVU/N/2cfnJR5SXY=";
   };
 
+  patches = [
+    # Fix for aarch64 and non-4K pages. Remove once upgraded past 1.6.2.
+    (fetchpatch {
+      url = "https://github.com/libunwind/libunwind/commit/e85b65cec757ef589f28957d0c6c21c498a03bdf.patch";
+      sha256 = "sha256-z2Hpg98D4UMmrE/LC596RFcyxRTvDjD4k7llDPfz1NI=";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libunwind/default.nix:16]:
```suggestion
      sha256 = "1lnlygvhqrdrjgw303pg2k2k4ms4gaghpjsgmhk47q83vy1yjwfg";
```

**Pull request merged**